modules/ding_place2book/ding_place2book.module

```diff
@@ -1,1050 +1,222 @@
-    $event_category_result = db_query('SELECT tid, name
-                                FROM {taxonomy_term_data}
-                                WHERE tid = :term_id', array(':term_id' => $field_ding_event_category[0]['tid']));
-    foreach ($event_category_result as $event_category) {
-      $event_category_tid = $event_category->tid;
-      $event_category_name = $event_category->name;
-    }
-
-    //check configuration if we should map the event category to a kultunaut category
-    $category_map_array = variable_get('ding_place2book_category_mappings', array());
-    if (empty($category_map_array[$event_category_tid])) {
-      $xml->event->kultunaut_export_category = $event_category_name;
+      $options[$name] = (string) drupal_array_get_nested_value($data, $option);
     }
-    else {
-      $xml->event->kultunaut_export_category = $category_map_array[$event_category_tid];
+    if (in_array($name, ['begin_at', 'end_at', 'sale_open_at', 'sale_close_at'])) {
+      array_pop($option);
+      $option[] = 'timezone_db';
+      $options[$name] .= (string) drupal_array_get_nested_value($data, $option);
     }
   }
-
-  // Set kultunaut picture URL
-  $field_ding_event_list_image = field_get_items('node', $node, 'field_ding_event_list_image');
-  //the uri of the image file is not found in the updated node object - we retrieve it with file_load
-  if (!empty($field_ding_event_list_image)) {
-    $list_image_fileinfo = file_load($field_ding_event_list_image[0]['fid']);
-    if (!empty($list_image_fileinfo->uri)) {
-      //this module provides a style, ding_place2book_kultunaut_square, which we invoke here
-      $xml->event->kultunaut_picture_uri = image_style_url('ding_place2book_kultunaut_square', $list_image_fileinfo->uri);
+  // @todo Find source for this param.
+  $options['address']['phone'] = 'phone';
+  if ($additional['kultunaut_export']) {
+    $field = field_get_items('node', $node, 'field_ding_event_category');
+    if ($field) {
+      $additional['kultunaut_export_category'] = _ding_place2book_get_export_category($field[0]['tid']);
     }
   }
 
-  return $xml;
-}
-
-/**
- * Loads default settings, if the an event already exists with the nid given the
- * settings for the event is loaded.
- */
-function ding_place2book_settings($nid = NULL) {
-  // Get default values.
-  $settings = variable_get('ding_place2book_event_nodes', array());
-
-  // Check database for node specific settings, if nid is defined.
-  if ($nid) {
-    $result = db_query('SELECT * FROM {ding_place2book} WHERE nid = :nid', array(':nid' => $nid));
-    if ($result->rowCount() > 0) {
-      $settings = $result->fetchAssoc();
-    }
-    //add place2book_id to define it if no database settings found
-    else {
-      $settings['place2book_id'] = NULL;
-    }
-  //add place2book_id to define it if nid is not defined
-  }
-  else {
-    $settings['place2book_id'] = NULL;
-  }
-  return $settings;
+  $options += $additional;
```

Comment:
When using addition operator for arrays, best is to check whether the operand is an array, unless we are 100% sure that this variable is always an array. Since this easily throws fatal errors when something is wrong.